### DISCUSSION TITLE
How to get client connection status?

### DISCUSSION THREAD

**Discussion by ZootHii** [eclipse-milo/milo#1021] [General]
Is there a way to know the connection status like `opcUaClient.isConnected()`.

Thank you.

**Comment by kevinherron** [comment]
Use `OpcUaClient::addSessionActivityListener` to be notified when the session goes active/inactive.

There is no direct `isConnected` method because the client is, in theory, transport agnostic, and transports like HTTP or a future UDP-based tranpsort wouldn't ever be "connected" like a TCP connection would be. So instead the library reports whether you have an active session or not.

Once you call `OpcUaClient::connect` the library will always be working to stay connected with an active session, you are not required to do anything other than add a `SubscriptionListener` via `UaSubscriptionManager::addSubscriptionListener` so that you can re-create subscriptions after the client reconnects if the `onSubscriptionTransferFailed` callback is invoked.
